internal/sys/fs.go

```diff
@@ -363,6 +167,32 @@ func (c *FSContext) LookupReaddir(idx int32, f *FileEntry) (*Readdir, syscall.Er
 	}
 }
 
+// dotDirents returns "." and "..", where "." because wasi-testsuite does inode
+// validation.
+func dotDirents(f *FileEntry) ([]fsapi.Dirent, syscall.Errno) {
```

Comment:
function restored from a previous revision 34324031cb6d1f4b96ff706c72acd054fc70d6fd in wasi_snapshot_preview1/fs.go